client/app/pages/dashboards/hooks/useAddDashboardToRecentDashboards.js

```diff
@@ -0,0 +1,31 @@
+import { reduce, map, find } from "lodash";
+import { useCallback } from "react";
+
+export default function useAddDashboardToRecentDashboards(dashboardId) {
+  return useCallback(() => {
+    const currentRecentDashboards = localStorage.getItem("recentDashboards");
+    if (!currentRecentDashboards) {
+      localStorage.setItem("recentDashboards", JSON.stringify([{ id: dashboardId, priority: 1 }]));
```

Comment:
And another thought: we only add items, but what about eviction? If we don't remove items after some usage we will end up with hundreds if not thousands of items in these lists for some usrs.

Reply:
As i said to gabriel:
>So basically biggest priority implies in appearing first, every time i need to update the "recents" array, i check for the biggest value in the array, and give my new item the priority value of the biggest value + 1, that way i only need to sort the array when i'm displaying it
But we can do it only whit an array of ids, also i think that using only ids on the array it will be easier to read to be honest.. i will work on it.

Reply:
As i said to gabriel:

>So basically biggest priority implies in appearing first, every time i need to update the "recents" array, i check for the >biggest value in the array, and give my new item the priority value of the biggest value + 1, that way i only need to sort >the array when i'm displaying it
But we can do it only whit an array of ids, also i think that using only ids on the array it will be easier to read to be honest.. i will work on it.

Reply:
As i said to Gabriel:

>So basically biggest priority implies in appearing first, every time i need to update the "recents" array, i check for the biggest value in the array, and give my new item the priority value of the biggest value + 1, that way i only need to sort the array when i'm displaying it

But we can do it only whit an array of ids, also using only ids on the array it will be easier to read to be honest.. i will work on it.

Reply:
My suggestion is to store just array of ids, where first id is the most recent item, so order of IDs in the list will define order of items. And updating this list will look like this (pseudocode):
```js
function addToRecents(list, id) {
  exclude(list, id);
  prepend(list, id);
  return slice(list, someLimitValue); // keep only reasonable count of items in list
}
```

Reply:
What Levko suggested is exactly what I was thinking in https://github.com/getredash/redash/pull/5150/files#r484527405. The pseudocode makes it clearer 🙂

Reply:
@gabrieldutra @kravets-levko  Done!